Move Android Nnapi srcs from aten_native_cpu to aten_cpu

Amyh11325 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #63111
* __->__ #62919

Move Android NNAPI srcs (nnapi_bind.cpp, nnapi_wrapper.cpp, nnapi_model_loader.cpp) from aten_native_cpu to aten_cpu, so that later the NNAPI delegate's execution library can depend on it.

aten_native_cpu is built selectively per app, but the srcs have no selective components and are required for the NNAPI delegate library in D30259033. 

See Buck Dependencies: https://docs.google.com/document/d/17RuWkqWKCO6sc5fKzIDkGeNhhvMk7BvJOqeSnGsHZ8o/edit?usp=sharing

Differential Revision: [D30164867](https://our.internmc.facebook.com/intern/diff/D30164867/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30164867/)!

Commit:
Move Android Nnapi srcs from aten_native_cpu to aten_cpu

Move nnapi srcs (nnapi_bind.cpp, nnapi_wrapper.cpp, nnapi_model_loader.cpp) from aten_native_cpu to aten_cpu, so that later the Nnapi delegate's execution library can depend use it.

See Buck Dependencies: https://docs.google.com/document/d/17RuWkqWKCO6sc5fKzIDkGeNhhvMk7BvJOqeSnGsHZ8o/edit?usp=sharing

Differential Revision: [D30164867](https://our.internmc.facebook.com/intern/diff/D30164867/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30164867/)!

[ghstack-poisoned]

Commit:
Update on "Move Android Nnapi srcs from aten_native_cpu to aten_cpu"


Move nnapi srcs (nnapi_bind.cpp, nnapi_wrapper.cpp, nnapi_model_loader.cpp) from aten_native_cpu to aten_cpu, so that later the Nnapi delegate's execution library can depend use it.

See Buck Dependencies: https://docs.google.com/document/d/17RuWkqWKCO6sc5fKzIDkGeNhhvMk7BvJOqeSnGsHZ8o/edit?usp=sharing

Differential Revision: [D30164867](https://our.internmc.facebook.com/intern/diff/D30164867/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30164867/)!

[ghstack-poisoned]

iseeyuan reviewed (commented):
This move should not change the production build essentially, because there is no selective components in nnapi runtime code. I'm running build size test and will stamp this PR when we get the results.

Commit:
Update on "Move Android Nnapi srcs from aten_native_cpu to aten_cpu"


Move nnapi srcs (nnapi_bind.cpp, nnapi_wrapper.cpp, nnapi_model_loader.cpp) from aten_native_cpu to aten_cpu, so that later the Nnapi delegate's execution library can depend use it.

See Buck Dependencies: https://docs.google.com/document/d/17RuWkqWKCO6sc5fKzIDkGeNhhvMk7BvJOqeSnGsHZ8o/edit?usp=sharing

Differential Revision: [D30164867](https://our.internmc.facebook.com/intern/diff/D30164867/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30164867/)!

[ghstack-poisoned]

Commit:
Update on "Move Android Nnapi srcs from aten_native_cpu to aten_cpu"


Move nnapi srcs (nnapi_bind.cpp, nnapi_wrapper.cpp, nnapi_model_loader.cpp) from aten_native_cpu to aten_cpu, so that later the Nnapi delegate's execution library can depend use it.

See Buck Dependencies: https://docs.google.com/document/d/17RuWkqWKCO6sc5fKzIDkGeNhhvMk7BvJOqeSnGsHZ8o/edit?usp=sharing

Differential Revision: [D30164867](https://our.internmc.facebook.com/intern/diff/D30164867/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30164867/)!

[ghstack-poisoned]

Commit:
Update on "Move Android Nnapi srcs from aten_native_cpu to aten_cpu"


Move nnapi srcs (nnapi_bind.cpp, nnapi_wrapper.cpp, nnapi_model_loader.cpp) from aten_native_cpu to aten_cpu, so that later the Nnapi delegate's execution library can depend use it.

See Buck Dependencies: https://docs.google.com/document/d/17RuWkqWKCO6sc5fKzIDkGeNhhvMk7BvJOqeSnGsHZ8o/edit?usp=sharing

Differential Revision: [D30164867](https://our.internmc.facebook.com/intern/diff/D30164867/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30164867/)!

[ghstack-poisoned]

Commit:
Update on "Move Android Nnapi srcs from aten_native_cpu to aten_cpu"


Move nnapi srcs (nnapi_bind.cpp, nnapi_wrapper.cpp, nnapi_model_loader.cpp) from aten_native_cpu to aten_cpu, so that later the Nnapi delegate's execution library can depend use it.

See Buck Dependencies: https://docs.google.com/document/d/17RuWkqWKCO6sc5fKzIDkGeNhhvMk7BvJOqeSnGsHZ8o/edit?usp=sharing

Differential Revision: [D30164867](https://our.internmc.facebook.com/intern/diff/D30164867/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30164867/)!

[ghstack-poisoned]

iseeyuan reviewed (approved):
LGTM

Commit:
Update on "Move Android Nnapi srcs from aten_native_cpu to aten_cpu"


Move Android NNAPI srcs (nnapi_bind.cpp, nnapi_wrapper.cpp, nnapi_model_loader.cpp) from aten_native_cpu to aten_cpu, so that later the NNAPI delegate's execution library can depend on it.

aten_native_cpu is built selectively per app, but the srcs have no selective components and are required for the NNAPI delegate library in D30259033. 

See Buck Dependencies: https://docs.google.com/document/d/17RuWkqWKCO6sc5fKzIDkGeNhhvMk7BvJOqeSnGsHZ8o/edit?usp=sharing

Differential Revision: [D30164867](https://our.internmc.facebook.com/intern/diff/D30164867/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30164867/)!

[ghstack-poisoned]

Commit:
Update on "Move Android Nnapi srcs from aten_native_cpu to aten_cpu"


Move Android NNAPI srcs (nnapi_bind.cpp, nnapi_wrapper.cpp, nnapi_model_loader.cpp) from aten_native_cpu to aten_cpu, so that later the NNAPI delegate's execution library can depend on it.

aten_native_cpu is built selectively per app, but the srcs have no selective components and are required for the NNAPI delegate library in D30259033. 

See Buck Dependencies: https://docs.google.com/document/d/17RuWkqWKCO6sc5fKzIDkGeNhhvMk7BvJOqeSnGsHZ8o/edit?usp=sharing

Differential Revision: [D30164867](https://our.internmc.facebook.com/intern/diff/D30164867/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30164867/)!

[ghstack-poisoned]

Commit:
Update on "Move Android Nnapi srcs from aten_native_cpu to aten_cpu"


Move Android NNAPI srcs (nnapi_bind.cpp, nnapi_wrapper.cpp, nnapi_model_loader.cpp) from aten_native_cpu to aten_cpu, so that later the NNAPI delegate's execution library can depend on it.

aten_native_cpu is built selectively per app, but the srcs have no selective components and are required for the NNAPI delegate library in D30259033. 

See Buck Dependencies: https://docs.google.com/document/d/17RuWkqWKCO6sc5fKzIDkGeNhhvMk7BvJOqeSnGsHZ8o/edit?usp=sharing

Differential Revision: [D30164867](https://our.internmc.facebook.com/intern/diff/D30164867/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30164867/)!

[ghstack-poisoned]
